src/repr/scalar/strconv.rs

```diff
@@ -200,6 +200,14 @@ fn parse_timestamp_string(s: &str) -> Result<(NaiveDate, NaiveTime, i64), failur
 
     let (ts_string, tz_string) = crate::datetime::split_timestamp_string(s);
 
+    // Reformat date strings into desired form of 'YYYY-MM-DD', including:
+    // - YYYY MM-DD
+    // - YYYY MM DD
+    // - YYYYMMDD
+    let date_formatter =
+        regex::Regex::new(r"^(?P<y>\d{4})\s*?(?P<m>\d{2})(?:\-|\s+)?(?P<d>\d{2})").unwrap();
```

Comment:
regex's [should be stuck in a lazy static](https://docs.rs/regex/1.3.7/regex/#example-avoid-compiling-the-same-regex-in-a-loop), even inside a function. They are very expensive to build, and unlike other languages they aren't cached in rust:

```suggestion
lazy_static! {
    static ref DATE_FORMATTER: Regex =
        regex::Regex::new(r"^(?P<y>\d{4})\s*?(?P<m>\d{2})(?:\-|\s+)?(?P<d>\d{2})").unwrap();
}
```